src/connection_state_machine.c

```diff
@@ -170,6 +171,12 @@ enum OperationReturnCode csm_next_state(struct state_machine_ctx_t *ctx)
             csm_set_state(ctx, LDAP_CONNECTION_STATE_INIT);
         }
         break;
+
+    case LDAP_CONNECTION_STATE_LOAD_SCHEMA:
+        rc = ldap_schema_load(ctx->ctx, ctx->ctx->schema);
+        csm_set_state(ctx, LDAP_CONNECTION_STATE_RUN);
+        break;
+
```

Comment:
`LDAP_CONNECTION_STATE_LOAD_SCHEMA` should be placed before `LDAP_CONNECTION_STATE_RUN`.